### PR TITLE
literate-git-0.3.1 (new formula)

### PR DESCRIPTION
**Pull request by lukel97**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

See #50282

### CODE REVIEW

**Comment by bubba** [comment]
I guess git is a dependency of the pygit resource?

File: Formula/literate-git.rb [hunk 1]
```diff
@@ -0,0 +1,78 @@
+class LiterateGit < Formula
+  include Language::Python::Virtualenv
+
+  desc "Render hierarchical git repositories into HTML"
+  homepage "https://github.com/bennorth/literate-git"
+  url "https://github.com/bennorth/literate-git/archive/v0.3.1.tar.gz"
+  sha256 "f1dec77584236a5ab2bcee9169e16b5d976e83cd53d279512136bdc90b04940a"
+  head "https://github.com/bennorth/literate-git.git"
+
+  depends_on "python"
```
Comment (SMillerDev) [inline Formula/literate-git.rb]: Does it work with `python@3.8`?

**Comment by bubba** [comment]
Looks like the pygit resource can't find `libgit2`, but it seems to already be an implicit dependency of the formula:

```
$ brew deps literate-git
gdbm
libgit2
openssl@1.1
python@3.8
readline
sqlite
xz
```

Do I need to explicitly specify it for the resource to pick it up?

**Comment by SMillerDev** [comment]
I don't think so, but you can try.

**Comment by bubba** [comment]
After uninstalling libgit2 on my machine, I was able to recreate the failure the buildbot was having.
I'm not sure where those extra dependencies are coming from in `brew deps`, but explicitly specifying `libgit2` now seems to work

**Comment by SMillerDev** [comment]
It's now failing on `brew audit`

**Comment by bubba** [comment]
@SMillerDev audit should be passing now

**Review by zbeekman — CHANGES_REQUESTED** [review]
@bubba Please change the python dependency to "python@3.8". We (currently) won't accept new formula unless they can build against Python 3.8.

Also, please squash your commits and force push once you have done this. (You can delete all git commit message content beyond the first line saying `literate-git-0.3.1 (new formula)`

**Comment by bubba** [comment]
@zbeekman done, hopefully this builds!

**Review by zbeekman — APPROVED** [review]
~~LGTM~~

Edit: I found a potential issue with `head`. Please confirm that local testing has been done and there is a good reason to have it.

File: Formula/literate-git.rb [hunk 2]
```diff
@@ -0,0 +1,79 @@
+class LiterateGit < Formula
+  include Language::Python::Virtualenv
+
+  desc "Render hierarchical git repositories into HTML"
+  homepage "https://github.com/bennorth/literate-git"
+  url "https://github.com/bennorth/literate-git/archive/v0.3.1.tar.gz"
+  sha256 "f1dec77584236a5ab2bcee9169e16b5d976e83cd53d279512136bdc90b04940a"
+  head "https://github.com/bennorth/literate-git.git"
```
Comment (zbeekman) [inline Formula/literate-git.rb]: Gah, sorry, one more thing: Why did you add `head`? Have you tested this locally? My suspicion is that this will fail and should be removed unless you have a good reason (like upstream adding nightly testing via `brew install --HEAD literate-git`).

**Review by zbeekman — CHANGES_REQUESTED** [review]

File: Formula/literate-git.rb (hunk 2 above)
Comment (bubba) [reply Formula/literate-git.rb]: Wasn’t there for any particular reason, happy to remove it!

**Review by zbeekman — APPROVED** [review]
LGTM!

Thanks again!

**Comment by zbeekman** [comment]
Thanks for your contribution to Homebrew! 🎉 🥇 

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you @bubba!
